### PR TITLE
School Library: decorating the class

### PR DESCRIPTION
**Pull request by Jeddaa**
- created the decorate class
- created the namerable class
- created the capitalize and trimmer decorate class
- added appropriate inheritance

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative_path 'decorator'
```
Comment (prolajumokeoni) [inline trimmer_decorator.rb]:
```suggestion
require_relative 'decorator'
```

- [ ] Kindly change the path to `require_relative 'decorator'`, you should use require_relative for local files 👍🏿

File: person.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
-class Person
+require_relative 'Nameable'
+
```
Comment (prolajumokeoni) [inline person.rb:4]: ### OPTIONAL 

- [ ] Kindly import `require_relative 'capitalize_decorator'
require_relative 'trimmer_decorator'` so you can test the example below ⬇️ 

```
person = Person.new(22, 'maximilianus')
p person.correct_name
 capitalizedPerson = CapitalizeDecorator.new(person)
p capitalizedPerson.correct_name
 capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
p capitalizedTrimmedPerson.correct_name
```

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️   Classes & method required in this project are   created 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Jeddaa 

Your project is complete! Great job🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Classes & methods ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**